### PR TITLE
Set C11 standard in CMake examples

### PR DESCRIPTION
**Pull request by hyoklee**
See also #4975.

### CODE REVIEW

**Review by derobins — APPROVED** [review]

File: HDF5Examples/config/cmake/HDFExampleMacros.cmake [hunk 1]
```diff
@@ -31,13 +31,13 @@ macro (BASIC_SETTINGS varname)
     set (CMAKE_MFC_FLAG 0)
   endif ()
 
-  set (CMAKE_C_STANDARD 99)
+  set (CMAKE_C_STANDARD 11)
```
Comment (byrnHDF) [inline HDF5Examples/config/cmake/HDFExampleMacros.cmake:34]: The examples need to work for all versions of the library - should this be conditional on the lib version?

File: config/cmake/HDFCompilerCXXFlags.cmake [hunk 2]
```diff
@@ -32,7 +32,7 @@ endif ()
 if (CMAKE_CXX_COMPILER_ID STREQUAL SunPro AND CMAKE_CXX_COMPILER_LOADED)
   if (NOT DEFINED CMAKE_CXX${CMAKE_CXX_STANDARD}_STANDARD_COMPILE_OPTION)
     if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.13)
-      if (NOT CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD EQUAL 98)
+      if (NOT CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD EQUAL 11)
```
Comment (byrnHDF) [inline config/cmake/HDFCompilerCXXFlags.cmake:35]: This is fine for the library.

**Review by byrnHDF — CHANGES_REQUESTED** [review]
Need to determine if this is a problem for older library versions.

File: HDF5Examples/config/cmake/HDFExampleMacros.cmake (hunk 1 above)
Comment (derobins) [reply HDF5Examples/config/cmake/HDFExampleMacros.cmake:34]: We can leave it at 99. I don't think the current examples do anything special for C11.

**Comment by fortnern** [comment]
Closed due to consensus at standup. Examples should not require C11

**Pull request closed without merging**